src/query/src/index.ts

```diff
@@ -248,6 +248,20 @@ export class Query {
     })
   }
 
+  /**
+   * Populates security insights for all nodes in the graph if a security
+   * archive is available. This ensures insights are available even when
+   * nodes haven't been returned by a query yet.
+   */
+  populateAllNodeInsights(): void {
+    if (!this.#securityArchive) {
+      return
+    }
+
+    const allNodes = new Set(this.#graph.nodes.values())
+    this.#getQueryResponseNodes(allNodes)
+  }
+
```

Comment:
if the goal is to always attach the insights to nodes it might be better to pull the logic that attach this info away from the `src/query` workspace and bring that as part of the `src/graph` workspace initialization/loading instead.